components/style/gecko/wrapper.rs

```diff
@@ -661,7 +661,8 @@ impl<'le> ::selectors::Element for GeckoElement<'le> {
             NonTSPseudoClass::MozTableBorderNonzero |
             NonTSPseudoClass::MozBrowserFrame => unsafe {
                 Gecko_MatchesElement(pseudo_class.to_gecko_pseudoclasstype().unwrap(), self.0)
-            }
+            },
+            NonTSPseudoClass::MozAny(ref selectors) => matches(&selectors.0, self, None),
```

Comment:
This doesn't pass the `ElementFlags`, nor the `StyleRelations` here. I believe it should be ok, because we unconditionally insert the appropriate flag for all the custom non-tree-structural pseudo-classes, and because of the limit on only-descendant combinators.

Can you verify that something like http://searchfox.org/mozilla-central/source/layout/reftests/css-selectors/state-dependent-in-any.html (perhaps tweaking manually with `:hover` or `:active`, or something we actually implement instead of `:valid`) works?